Content/Items/Manabonds/Accessories.Manabond.cs

```diff
@@ -0,0 +1,163 @@
+			if (owner?.HasMinionAttackTargetNPC ?? false)
+			{
+				NPC npc = Main.npc[owner.MinionAttackTargetNPC];
+
+				if (Vector2.Distance(minion.Center, npc.Center) <= range && Collision.CanHit(minion.Center, 1, 1, npc.Center, 1, 1))
+					return npc;
+			}
+
+			// Else, scan for other valid NPCs
+			foreach (NPC npc in Main.npc)
+			{
+				if (!npc.active || npc.friendly || !npc.CanBeChasedBy(this))
+					continue;
+
+				if (Vector2.Distance(minion.Center, npc.Center) <= range && Collision.CanHit(minion.Center, 1, 1, npc.Center, 1, 1))
+					return npc;
+			}
+
+			return null;
+		}
+
+		public override bool SafeCanAccessoryBeEquippedWith(Item equipped, Item incoming, Player player)
+		{
+			if (equipped.ModItem is Manabond && incoming.ModItem is Manabond)
+				return false;
+
+			return true;
+		}
+	}
+
+	/// <summary>
+	/// This GlobalProjectile handles the mana system and AI for minions under the effect of a manabond accessory.
+	/// This should only ever do anything if a minion's owner has a valid manabond equipped.
+	/// </summary>
+	internal class ManabondProjectile : GlobalProjectile
+	{
+		public int mana;
+		public int maxMana = 40;
+
+		public int timer = 0;
+
+		public NPC target;
+
+		public override bool InstancePerEntity => true;
+
+		public override bool AppliesToEntity(Projectile entity, bool lateInstantiation)
+		{
+			return entity.minion;
+		}
+
+		/// <summary>
+		/// Determines if this minion's owner has a manabond on, and thus should have their mana siphoned
+		/// </summary>
+		/// <param name="projectile">The minion to check the owner's status of</param>
+		/// <returns>If the minions owner has a manabond equipped</returns>
+		public bool ManaActive(Projectile projectile)
+		{
+			Player owner = Main.player[projectile.owner];
+			return owner.armor.Any(n => n.ModItem is Manabond); // Checks if the player has any manabond on
+		}
+
+		/// <summary>
+		/// Retrieves the currently equipped manabond of the given projectile's owner, or null if one is not equipped
+		/// </summary>
+		/// <param name="projectile">The minion to check the owner's manabond of</param>
+		/// <returns>The equipped manabond if one is equipped, or null if not</returns>
+		public Manabond GetActiveBond(Projectile projectile)
+		{
+			if (!ManaActive(projectile))
+				return null;
+
+			Player owner = Main.player[projectile.owner];
+
+			return owner.armor.FirstOrDefault(n => n.ModItem is Manabond).ModItem as Manabond;
+		}
+
+		public override void PostAI(Projectile projectile)
+		{
+			if (!ManaActive(projectile))
+				return;
+
+			Manabond bond = GetActiveBond(projectile);
+			Player owner = Main.player[projectile.owner];
+
+			timer++;
+
+			// Drop target attempt
+			if (target != null && (!target.active || Vector2.Distance(projectile.Center, target.Center) > 800))
+				target = null;
+
+			// Retarget attempt
+			target ??= bond?.FindTarget(projectile, this);
+
+			// Mana siphon attempt
```

Comment:
should probably favor Owner.CheckMana() over decreasing manually. See https://github.com/ProjectStarlight/StarlightRiver/blob/bee420ed93f18e4692fe0ff5c4de127dab5cace1/Content/Items/UndergroundTemple/Weapons.RuneStaff.cs#L125C2-L125C2